src/integrations/pathManager.ts

```diff
@@ -0,0 +1,108 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import paths from '@/utils/paths';
+import { manageLinesInFile } from '@/integrations/manageLinesInFile';
+
+let pathManager: ManualPathManager;
+
+// PathManager is the interface that anything that manages the
+// PATH variable must implement.
+interface PathManager {
+  // Makes real any changes to the system. Should be idempotent.
+  enforce(): Promise<void>
+  // Removes any changes that the PathManager may have made.
+  // Should be idempotent.
+  remove(): Promise<void>
+}
+
+// ManualPathManager is for when the user has chosen to manage
+// their PATH themselves. It does nothing.
+class ManualPathManager implements PathManager {
+  async enforce(): Promise<void> {}
+  async remove(): Promise<void> {}
+}
+
+// RcFilePathManager is for when the user wants Rancher Desktop to
+// make changes to their PATH by putting lines that change it in their
+// shell .rc files.
+export class RcFilePathManager implements PathManager {
+  constructor() {
+    const platform = os.platform();
+
+    if (platform !== 'linux' && platform !== 'darwin') {
+      throw new Error(`Platform "${ platform }" is not supported by RcFilePathManager`);
+    }
+  }
+
+  async enforce(): Promise<void> {
+    await this.managePosix(true);
+    await this.manageCsh(true);
+    await this.manageFish(true);
+  }
+
+  async remove(): Promise<void> {
+    await this.managePosix(false);
+    await this.manageCsh(false);
+    await this.manageFish(false);
+  }
+
+  protected async managePosix(desiredPresent: boolean): Promise<void> {
+    const pathLine = `PATH=\${PATH}:${ paths.integration }`;
+
+    await Promise.all(['.bashrc', '.zshrc'].map((rcName) => {
+      const rcPath = path.join(os.homedir(), rcName);
+
+      return manageLinesInFile(rcPath, [pathLine], desiredPresent);
+    }));
+  }
+
+  protected async manageCsh(desiredPresent: boolean): Promise<void> {
+    const pathLine = `set path=($path ${ paths.integration })`;
+
+    await Promise.all(['.cshrc', '.tcshrc'].map((rcName) => {
+      const rcPath = path.join(os.homedir(), rcName);
+
+      return manageLinesInFile(rcPath, [pathLine], desiredPresent);
+    }));
+  }
+
+  protected async manageFish(desiredPresent: boolean): Promise<void> {
+    const pathLine = `set -x PATH "${ paths.integration }" "$PATH"`;
```

Comment:
There appears to be a [`fish_add_path`](https://fishshell.com/docs/current/cmds/fish_add_path.html); would that be useful here?

If not, we should use long form here (`--export` instead of `-x`).

I'm not certain, but there appears to be an `--append` switch to modify the value in-place instead.

Speaking of, this appears to _prepend_ our path, whereas we _append_ in the other forms (I think?). Is that intended?

Reply:
Looks like `fish_add_path` [was added in fish 3.2.0](https://fishshell.com/docs/current/relnotes.html). I think that's too recent; for example I get version 3.1.0 when I install via `apt`. Otherwise it would be great.

> If not, we should use long form here (--export instead of -x).
>
> I'm not certain, but there appears to be an --append switch to modify the value in-place instead.

Done, we're using `--export` and `--append` now.

> Speaking of, this appears to prepend our path, whereas we append in the other forms (I think?). Is that intended?

No it wasn't, nice catch. I was working off of the python link in https://github.com/rancher-sandbox/rancher-desktop/issues/1719 and it looks like copy-paste got away on me :)